### PR TITLE
refactor: migrate loan transaction screen to compose

### LINKED ISSUE #2572: refactor: Loan Account Transaction Screen Migration to Compose
**Summary:**

Loan Account Transaction Screen Migration to Compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Fixes #2572 

Please Add Screenshots If there are any UI changes.

Please Add Screenshots If there are any UI changes.
| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| screenshot_20240425_020246_mifos_mobile_720 | screenshot_20240425_023123_mifos_mobile_720 |

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Created by dilpreet on 4/3/17.
+ */
+@AndroidEntryPoint
+class LoanAccountTransactionFragment : BaseFragment() {
+
+    private val viewModel: LoanAccountTransactionViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (activity as? BaseActivity)?.showToolbar()
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt]: can be moved to compose

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        (activity as BaseActivity).hideToolbar()
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt]: move this in a function

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            setContent {
+                MifosMobileTheme {
+                    LoanAccountTransactionScreen(
+                        navigateBack = { activity?.supportFragmentManager?.popBackStack() }
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt]: can it be replaced by `onBackPressedDispatcher.onBackPressed()` ?

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt (hunk 3 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt]: `activity?.supportFragmentManager?.popBackStack()`: Use this to programmatically navigate back (pop) through fragment transactions and manage the fragment back stack within an activity or fragment.

`onBackPressedDispatcher.onBackPressed()`: Use this to programmatically simulate the default back button behavior (like pressing the device's back button) from contexts where direct access to onBackPressed() of the activity is not available or convenient (e.g., within ViewModels, custom classes, etc.).

So, by these definitions, I believe we should go with pop backstack()? What are your views?

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private val _loanWithAssociations = MutableStateFlow<LoanWithAssociations>(LoanWithAssociations())
+    val loanWithAssociations: StateFlow<LoanWithAssociations> = _loanWithAssociations
+
+    private var _loanId: Long? = 0
+    private val loanId get() = _loanId
+
+    fun loadLoanAccountDetails() {
+        viewModelScope.launch {
+            _loanUiState.value = LoanAccountTransactionUiState.Loading
+            loanRepositoryImp.getLoanWithAssociations(Constants.TRANSACTIONS, loanId)
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionViewModel.kt]: use stateIn here

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    fun loadLoanAccountDetails() {
+        loanUiState = loanRepositoryImp.getLoanWithAssociations(Constants.TRANSACTIONS, loanId)
+            .map { loanWithAssociations ->
+                if (loanWithAssociations != null && !loanWithAssociations.transactions.isNullOrEmpty()) {
+                    LoanAccountTransactionUiState.Success(loanWithAssociations = loanWithAssociations)
+                } else {
+                    LoanAccountTransactionUiState.Error(isDataEmpty = true)
+                }
+            }
+            .catch {  emit(LoanAccountTransactionUiState.Error()) }
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionViewModel.kt]: You don't need to catch the error, it does out of box

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Pull request merged**